lib/iris/__init__.py

```diff
@@ -427,4 +428,10 @@ def load_raw(uris, constraints=None, callback=None):
 
 def sample_data_path(*path_to_join):
     """Given the sample data resource, returns the full path to the file."""
-    return os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    target = os.path.join(iris.config.SAMPLE_DATA_DIR, *path_to_join)
+    if not glob.glob(target):
```

Comment:
Yeah, I mucked around with this for a short while but the definition of "wrong" was somewhat ambiguous. For example, if I've installed the sample data at `/samples` then is it unambiguously wrong to do `sample_data_path('/samples/air_temp.pp')`? It's certainly against the _spirit_ of the function. It probably wouldn't take too much to persuade me to rule out _any_ absolute path - at which point we've probably caught most of the common misuse patterns.